frontend/web/components/EditPermissions.tsx

```diff
@@ -51,34 +45,40 @@ import {
 } from 'common/services/useRolePermissionGroup'
 
 import {
-  useGetUserWithRolesQuery,
   useDeleteUserWithRolesMutation,
+  useGetUserWithRolesQuery,
 } from 'common/services/useUserWithRole'
 
 import {
-  useGetGroupWithRoleQuery,
   useDeleteGroupWithRoleMutation,
+  useGetGroupWithRoleQuery,
 } from 'common/services/useGroupWithRole'
 
 import MyRoleSelect from './MyRoleSelect'
 import { setInterceptClose } from './modals/base/ModalDefault'
+import Panel from './base/grid/Panel'
+import InputGroup from './base/forms/InputGroup'
+
 const OrganisationProvider = require('common/providers/OrganisationProvider')
 const Project = require('common/project')
 
 type EditPermissionModalType = {
   group?: UserGroup
-  id: string
+  id: number
   isGroup?: boolean
   level: PermissionLevel
   name: string
   onSave: () => void
+  envId?: number
   parentId?: string
   parentLevel?: string
   parentSettingsLink?: string
+  roleTabTitle: string
   permissions?: UserPermission[]
   push: (route: string) => void
   user?: User
   role?: Role
+  roles?: Role[]
```

Comment:
It reads kinda weird that both role and roles are present on `EditPermissionModalType`.